diff -crN igo-0.4.2-src/src/net/reduls/igo/dictionary/WordDic.java igo-0.4.2-gae/src/net/reduls/igo/dictionary/WordDic.java
*** igo-0.4.2-src/src/net/reduls/igo/dictionary/WordDic.java	2010-03-29 09:25:40.000000000 +0900
--- igo-0.4.2-gae/src/net/reduls/igo/dictionary/WordDic.java	2011-03-20 22:22:26.649173305 +0900
***************
*** 17,23 ****
  
      public WordDic(String dataDir) throws IOException {
  	trie    = new Searcher(dataDir+"/word2id");
! 	data    = FileMappedInputStream.getString(dataDir+"/word.dat");
  	indices = FileMappedInputStream.getIntArray(dataDir+"/word.ary.idx");
  	
  	{
--- 17,25 ----
  
      public WordDic(String dataDir) throws IOException {
  	trie    = new Searcher(dataDir+"/word2id");
! 	data    = FileMappedInputStream.getString(dataDir+"/word.dat.1")+
!                   FileMappedInputStream.getString(dataDir+"/word.dat.2")+
!                   FileMappedInputStream.getString(dataDir+"/word.dat.3");
  	indices = FileMappedInputStream.getIntArray(dataDir+"/word.ary.idx");
  	
  	{
diff -crN igo-0.4.2-src/src/net/reduls/igo/dictionary/build/WordDic.java igo-0.4.2-gae/src/net/reduls/igo/dictionary/build/WordDic.java
*** igo-0.4.2-src/src/net/reduls/igo/dictionary/build/WordDic.java	2010-03-29 18:35:02.000000000 +0900
--- igo-0.4.2-gae/src/net/reduls/igo/dictionary/build/WordDic.java	2011-03-20 22:22:26.649173305 +0900
***************
*** 131,143 ****
  	}
  
  	// 単語データを出力
! 	final FileMappedOutputStream fmosDat = 
! 	    new FileMappedOutputStream(outputDir+"/word.dat", wdat.length()*2);
! 	try {
! 	    fmosDat.putString(wdat.toString());
! 	} finally {
! 	    fmosDat.close();
! 	}
  	
  	// 単語情報の配列へのインデックスを保存する
  	{ 
--- 131,147 ----
  	}
  
  	// 単語データを出力
!         final int[] wdatLens = {wdat.length()/3, wdat.length()/3, wdat.length()/3+wdat.length()%3};
!         final int[] wdatBegs = {0, wdat.length()/3, wdat.length()/3*2};
!         for(int i=0; i < 3; i++) {
!             final FileMappedOutputStream fmosDat = 
!                 new FileMappedOutputStream(outputDir+"/word.dat."+(i+1), wdatLens[i]*2);
!             try {
!                 fmosDat.putString(wdat.substring(wdatBegs[i],wdatBegs[i]+wdatLens[i]).toString());
!             } finally {
!                 fmosDat.close();
!             }
!         }
  	
  	// 単語情報の配列へのインデックスを保存する
  	{ 
diff -crN igo-0.4.2-src/src/net/reduls/igo/util/FileMappedInputStream.java igo-0.4.2-gae/src/net/reduls/igo/util/FileMappedInputStream.java
*** igo-0.4.2-src/src/net/reduls/igo/util/FileMappedInputStream.java	2010-03-29 18:35:02.000000000 +0900
--- igo-0.4.2-gae/src/net/reduls/igo/util/FileMappedInputStream.java	2011-03-20 23:25:20.548548785 +0900
***************
*** 3,9 ****
  import java.io.IOException;
  import java.io.FileInputStream;
  import java.nio.ByteBuffer;
- import java.nio.ByteOrder;
  import java.nio.channels.FileChannel;
  
  /**
--- 3,8 ----
***************
*** 12,18 ****
   */
  public final class FileMappedInputStream {
      private final FileChannel cnl;
!     private int cur=0;
  
      /**
       * 入力ストリームを作成する
--- 11,17 ----
   */
  public final class FileMappedInputStream {
      private final FileChannel cnl;
!     private final ByteBuffer buf;
  
      /**
       * 入力ストリームを作成する
***************
*** 21,35 ****
       */
      public FileMappedInputStream(String filepath) throws IOException {
  	cnl = new FileInputStream(filepath).getChannel();
      }
  
      public int getInt() throws IOException {
! 	return map(4).getInt();
      }
      
      public int[] getIntArray(int elementCount) throws IOException {
  	final int[] ary = new int[elementCount];
! 	map(elementCount*4).asIntBuffer().get(ary);
  	return ary;
      }
      
--- 20,38 ----
       */
      public FileMappedInputStream(String filepath) throws IOException {
  	cnl = new FileInputStream(filepath).getChannel();
+         buf = ByteBuffer.allocateDirect(size());
+         cnl.read(buf);
+         buf.flip();
      }
  
      public int getInt() throws IOException {
! 	return buf.getInt();
      }
      
      public int[] getIntArray(int elementCount) throws IOException {
  	final int[] ary = new int[elementCount];
! 	buf.asIntBuffer().get(ary);
!         increment(elementCount*4);
  	return ary;
      }
      
***************
*** 44,61 ****
  
      public short[] getShortArray(int elementCount) throws IOException {
  	final short[] ary = new short[elementCount];
! 	map(elementCount*2).asShortBuffer().get(ary);
  	return ary;
      }
  
      public char[] getCharArray(int elementCount) throws IOException {
  	final char[] ary = new char[elementCount];
! 	map(elementCount*2).asCharBuffer().get(ary);
  	return ary;
      }
  
      public String getString(int elementCount) throws IOException {
! 	return map(elementCount*2).asCharBuffer().toString();
      }
  
      public static String getString(String filepath) throws IOException {
--- 47,68 ----
  
      public short[] getShortArray(int elementCount) throws IOException {
  	final short[] ary = new short[elementCount];
! 	buf.asShortBuffer().get(ary);
!         increment(elementCount*2);
  	return ary;
      }
  
      public char[] getCharArray(int elementCount) throws IOException {
  	final char[] ary = new char[elementCount];
! 	buf.asCharBuffer().get(ary);
!         increment(elementCount*2);
  	return ary;
      }
  
      public String getString(int elementCount) throws IOException {
! 	String tmp = buf.asCharBuffer().toString();
!         increment(elementCount*2);
!         return tmp;
      }
  
      public static String getString(String filepath) throws IOException {
***************
*** 77,84 ****
  	} catch (IOException e) {}
      }
  
!     private ByteBuffer map(int size) throws IOException {
! 	cur += size;
! 	return cnl.map(FileChannel.MapMode.READ_ONLY, cur-size, size).order(ByteOrder.nativeOrder());
      }
! }
\ ファイル末尾に改行がありません
--- 84,90 ----
  	} catch (IOException e) {}
      }
  
!     private void increment(int size) {
!         buf.position(buf.position()+size);
      }
! }
diff -crN igo-0.4.2-src/src/net/reduls/igo/util/FileMappedOutputStream.java igo-0.4.2-gae/src/net/reduls/igo/util/FileMappedOutputStream.java
*** igo-0.4.2-src/src/net/reduls/igo/util/FileMappedOutputStream.java	2010-03-29 18:35:02.000000000 +0900
--- igo-0.4.2-gae/src/net/reduls/igo/util/FileMappedOutputStream.java	2011-03-20 22:22:26.649173305 +0900
***************
*** 26,32 ****
  	final FileChannel cnl = new RandomAccessFile(filepath,"rw").getChannel();
  	try {
  	    mbb = cnl.map(FileChannel.MapMode.READ_WRITE, 0, size);
! 	    mbb.order(ByteOrder.nativeOrder());
  	} finally {
  	    cnl.close();
  	}
--- 26,32 ----
  	final FileChannel cnl = new RandomAccessFile(filepath,"rw").getChannel();
  	try {
  	    mbb = cnl.map(FileChannel.MapMode.READ_WRITE, 0, size);
! 	    mbb.order(ByteOrder.BIG_ENDIAN);
  	} finally {
  	    cnl.close();
  	}
